adapters/Vungle/VungleAdapter/GADMAdapterVungleDelegate.h

```diff
@@ -35,13 +33,20 @@ typedef NS_ENUM(NSUInteger, BannerRouterDelegateState) {
 - (void)adAvailable;
 - (void)adNotAvailable:(nonnull NSError *)error;
 - (void)willShowAd;
-- (void)willCloseAd:(BOOL)completedView didDownload:(BOOL)didDownload;
-- (void)didCloseAd:(BOOL)completedView didDownload:(BOOL)didDownload;
+- (void)willCloseAd;
+- (void)didCloseAd;
+- (void)trackClick;
 
 @optional
 
-// Get banner request object
-@property(nonatomic, nonnull) GADMAdapterVungleBannerRequest *bannerRequest;
+// Differentiate two banners with same placement ID
+@property(nonatomic, nullable) NSString *uniquePubRequestID;
+
+// Reward user for watching a Rewarded Video Ad successfully
+- (void)rewardUser;
+
+// App will leave current application
+- (void)willLeaveApplication;
```

Comment:
Can we make these required and have the classes conforming too this delegate just do nothing? That way we need less "if respondsToSelector" calls.

Reply:
@ericleich Hi Eric, you mean we make `- (void)willLeaveApplication` as required not for `- (void)rewardUser`, right? Because I think `- (void)rewardUser` is only for Rewarded Video ad ? Thanks a lot. ^_^

Reply:
@YueVungle I'd like to make both required for every adapter to implement. It's a side effect of every adapter type using the same GADMAdapterVungleDelegate. Agree that banner/interstitial will have an empty implementation for `- (void)rewardUser` but I'd prefer that as it's more explicit that the adapter is taking no action. It also enforces that rewarded adapters have to implement rewardUser, for example, instead of omitting it. I feel the more explicit no-op avoids future bugs. Additionally, we can then get rid of `if respondsToSelector:` checks.

Reply:
@ericleich Hi Eric, thanks a lot for the detailed explanation, I will make `- (void)willLeaveApplication` and `- (void)rewardUser` as required. ^_^

Reply:
@ericleich Updated, thanks. 😊